opentreemap/treemap/tests/test_auth.py

```diff
@@ -54,6 +55,16 @@ def test_password_incorrect(self):
     #     res = self.client.get('/accounts/profile/')
     #     self.assertRedirects(res, '/accounts/profile/login/')
 
+    @override_settings(SOCIAL_AUTH_GOOGLE_OAUTH2_KEY='1', SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET='2')
+    @mock.patch('social_core.backends.oauth.OAuthAuth.state_token', return_value='zz')
+    def test_google_auth_redirect(self, *args):
+        response = self.client.get('/oauth/login/google-oauth2/')
+        self.assertRedirects(
+            response,
+            'https://accounts.google.com/o/oauth2/auth?scope=openid+email+profile&state=zz&redirect_uri=http://testserver/oauth/complete/google-oauth2/&response_type=code&client_id=1',
```

Comment:
I'll reformat this. This url is used because there is problem with mocking google api. At least I failed to do this.